services/QuillLMS/app/controllers/teachers/classroom_manager_controller.rb

```diff
@@ -133,20 +134,50 @@ def scores
   end
 
   def my_account
-    @time_zones = [{name: 'Select Time Zone', id: 'Select Time Zone'}].concat(TZInfo::Timezone.all_country_zone_identifiers.sort.map{|tz| {name: tz.gsub('_', ' '), id: tz}})
+    session[:google_redirect] = request.env['PATH_INFO']
+    session[:clever_redirect] = request.env['PATH_INFO']
+    @google_or_clever_just_set = session[:google_or_clever_just_set]
+    session[:google_or_clever_just_set] = nil
+    @account_info = get_account_info
   end
 
-  def my_account_data
-    render json: current_user.generate_teacher_account_info
+  def update_my_account
+    response = current_user.update_teacher(params['classroom_manager'])
```

Comment:
As discussed, this method might be better in the model and should use the standard rails errors.

Reply:
I agree with this note, but if it's okay with you, I'm going to leave it as a @TODO for myself because we use the current rails errors in multiple places and I don't want to rush testing a change in authentication.

Reply:
:+1: